lib/jasmine/jasmine.ts

```diff
@@ -89,10 +89,10 @@
     // The `done` callback is only passed through if the function expects at least one argument.
     // Note we have to make a function with correct number of arguments, otherwise jasmine will
     // think that all functions are sync or async.
-    return (testBody.length == 0) ? function() {
-      return testProxyZone.run(testBody, this);
-    } : function(done) {
+    return (testBody && testBody.length) ? function(done) {
       return testProxyZone.run(testBody, this, [done]);
+    } : function() {
```

Comment:
If we always return a function, won't it essentially make all specs non-pending (since there will be a callback specified)?
I think this should be:

```ts
return testBody && ((testBody.length == 0) ? function() { ... } : function { ... });
```